bin/pytorch_inference/CThreadSettings.cc

```diff
@@ -19,9 +18,25 @@
 namespace ml {
 namespace torch {
 
-void validateThreadingParameters(std::int32_t maxThreads,
-                                 std::int32_t& inferenceThreads,
-                                 std::int32_t& modelThreads) {
+CThreadSettings::CThreadSettings(std::int32_t inferenceThreads, std::int32_t numAllocations)
+    : m_InferenceThreads(inferenceThreads), m_NumAllocations(numAllocations) {
+}
+
+std::int32_t CThreadSettings::inferenceThreads() const {
+    return m_InferenceThreads;
+}
+
+std::int32_t CThreadSettings::numAllocations() const {
+    return m_NumAllocations;
+}
+
+void CThreadSettings::numAllocations(std::int32_t numAllocations) {
+    m_NumAllocations = numAllocations;
+}
+
+void CThreadSettings::validateThreadingParameters(std::int32_t maxThreads,
```

Comment:
This logic needs reviewing

Reply:
It's less important what the exact logic is if the Java side is going to protect us from exceeding these limits in the normal case, and these checks are only for the unfortunate case of ML nodes being lost.

Reply:
@dimitris-athanasiou what do you think? Should we delete this code now or keep a simplified version of the logic that rejects thread settings > hardware concurrency